internal/pkg/state/view.go

```diff
@@ -403,19 +403,11 @@ func (v *View) MarketHasDealID(ctx context.Context, addr addr.Address, dealID ab
 	}
 
 	found := false
-	byParty, err := market.AsSetMultimap(StoreFromCbor(ctx, v.ipldStore), marketState.DealIDsByParty)
+	dealStates, err := v.asDealStateArray(ctx, marketState.States)
```

Comment:
@anorth I think that this will achieve the same result but I'm going off of an uninformed interpretation of market actor state names, so you should double check that this is ok

Reply:
I don't know why this was done in such a complicated way in the first place.